UdpVideo/UdpVideoController.cpp

```diff
@@ -0,0 +1,169 @@
+#include "UdpVideoController.h"
+#include <windows.h>
+#include <libloaderapi.h>
+#include <QFile>
+#include <iostream>
+#include <QCoreApplication>
+
+namespace
+{
+extern "C"
+{
+enum GstState
+{
+    GST_STATE_NULL = 1,
+    GST_STATE_PLAYING = 4
+};
+enum GstStateChangeReturn
+{
+    last = 3
+};
+
+using GstElement = void;
+using GError = void;
+int (*gst_init)(int*,char***) = nullptr;
+GstElement* (*gst_parse_launch)(const char*, GError**) = nullptr;
+GstStateChangeReturn (*gst_element_set_state)(GstElement*, GstState) = nullptr;
+void (*gst_object_unref)(void*) = nullptr;
+} //extern C
+} //anon ns
+
+void check(const char* dll)
+{
+    if(NULL == LoadLibraryEx(dll, NULL, LOAD_LIBRARY_SEARCH_USER_DIRS))
+    {
+        auto error = GetLastError();
+        std::cerr << "Failed to load: " << dll << " : " << error << std::endl;
+    }
+}
+
+struct UdpVideoController::Pimpl

```

Comment:
I would extract it to separate file, it's quite hard to read now.

Reply:
splitted